### ISSUE TITLE
Curtains status not working

### ISSUE THREAD

**Issue by a7mad3bdullah** [tuya/tuya-homebridge#158]
**Describe the bug**
The plugin v1.5.0 has been added to homebridge successfully and authorization also works fine. But I’ve added a curtain to my account and been trying to control it from Home App, it keeps showing opening/closing even if the action has been done successfully (see screenshots). It also sometimes shows the status as open while it is closed and the other way around. 

**Expected behavior**

1- The status of the curtain should show properly as per the device status (open/closed)

2- When a command is executed, the device should report prober status to homebridge and that will be reported to Home App

**Screenshots**
![4DC4762B-F305-487A-9522-AE7A12CC9D3F](https://user-images.githubusercontent.com/44746139/141696675-934c2011-3431-4fec-9497-2d8073865d6d.png)
![C107758A-3D1D-4432-8537-DF33D4898046](https://user-images.githubusercontent.com/44746139/141696676-9bbb5b04-2c6d-4903-970f-c812a2db16ef.png)
![01D25F71-A60A-4E71-B7E7-6FE54BBC8905](https://user-images.githubusercontent.com/44746139/141696674-9ef288fa-f1da-4912-8768-4b6cc9145e3b.png)

**Device info**
can be provided if needed.

**Comment by wveilleux** [comment]
I have exactly the same issue. While the blinds are 100% open, either Home app and Homebridge UI shows the blinds in a "Closing" state. I've seen no errors in the log file. Any clue how to resolve that issue ?

**Comment by jmcollin78** [comment]
Hello same for me. The Smart Life application, the curtain itself are open (resp. closed), but in Home Assistant the curtains are displayed as closed (reps. open) and action to open (resp. close) is proposed.

Only the Home Assistant plugin seems to have reverse the state.

**Comment by jmcollin78** [comment]
When looking at the code it seems reversed here:

https://github.com/tuya/tuya-homebridge/blob/master/lib/window_covering_accessory.js line 30

```
refreshAccessoryServiceIfNeed(stateArr, isRefresh) {
        this.isRefresh = isRefresh;
        for (const statusMap of stateArr) {

            //Check whether 100% is fully on or fully off. If there is no dp point, 100% is fully off by default
            if (statusMap.code === 'situation_set') {
                this.fullySituationMap = statusMap
            }
```


100% means 100% open or fully on.

**Comment by reidcooper** [comment]
I am running into something similar. 

For example, if I shut my blinds from Apple Home, I can see that the device in Apple Home shows that it is closed. 

In Tuya, I can see that the device is closed.

That's great.

But, if I manually open up my blinds by pressing the device button, I can see the logging MQTT messages from the tuya-homebridge plugin. **However, the plugin will not notify Homebridge and Apple Home that the blinds are opened to 30%.**

**MQTT Messages I receive when the blinds are set to 89%**

```
TuyaOpenMQ onMessage: topic = cloud/token/in/ASDF, message = {"data":{"dataId":"ASDF","devId":"ASDF","productKey":"ASDF","status":[{"3":99,"code":"percent_state","t":"1642296746","value":99}]},"protocol":4,"pv":"2.0","sign":"7cf4f8813fe2755c1efe9ba33b6b787a","t":1642296746}
TuyaOpenMQ onMessage: topic = cloud/token/in/ASDF, message = {"data":{"dataId":"ASDF","devId":"ASDF","productKey":"ASDF","status":[{"3":89,"code":"percent_state","t":"1642296749","value":89}]},"protocol":4,"pv":"2.0","sign":"dec26c416130c55f1b99985632e6dfb5","t":1642296749}
```

The only part that stands out to me differently is the `closing` state message. That could be why Homebridge continues to keep showing your Blinds as "closing". My is doing that too.

Here, my blinds are closed 100% by manually pressing the blinds button on the device.

<img width="172" alt="Screen Shot 2022-01-15 at 8 40 59 PM" src="https://user-images.githubusercontent.com/6921277/149644027-1eec494e-e90a-4ccf-9dcf-910ab1310329.png">
 
Homebridge sees the current position as 0% but "Closing". Apple Home sees the blinds as "Closed".

**MQTT Messages I receive when the blinds are set to 100% closed**

```
TuyaOpenMQ onMessage: topic = cloud/token/in/ASDF, message = {"data":{"dataId":"ASDF","devId":"ASDF","productKey":"ASDF","status":[{"3":88,"code":"percent_state","t":"1642296801","value":88}]},"protocol":4,"pv":"2.0","sign":"312a3d83bdb52d60f5e60ae3e1e12d0d","t":1642296801}
TuyaOpenMQ onMessage: topic = cloud/token/in/ASDF, message = {"data":{"dataId":"ASDF","devId":"ASDF","productKey":"ASDF","status":[{"7":"closing","code":"work_state","t":"1642296802","value":"closing"}]},"protocol":4,"pv":"2.0","sign":"2f5c6eb7b50f086abc31e92ae0995db7","t":1642296802}
TuyaOpenMQ onMessage: topic = cloud/token/in/ASDF, message = {"data":{"dataId":"ASDF","devId":"ASDF","productKey":"ASDF","status":[{"3":100,"code":"percent_state","t":"1642296805","value":100}]},"protocol":4,"pv":"2.0","sign":"5fc79e758977dad6343cb97122842a51","t":1642296805}
```

**Comment by reidcooper** [comment]
I think this commit https://github.com/tuya/tuya-homebridge/commit/68684e3c58d8afc8242c2590c5e16ac4692005cf

removed one of the required characteristics according to Homebridge: Position State.

https://developers.homebridge.io/#/service/WindowCovering

I think because this was removed, it will no longer tell you if your Blinds are actually "done" moving so they will forever stay in the "Opening" or "Closing" state. But this is just a theory of mine.

**Comment by Geczy** [comment]
can you fix it?

**Comment by leo150** [comment]
Thank you for pointing the commit. Nasty little fix for whoever is interested (#363).
